README.md

```diff
@@ -920,6 +938,251 @@ branch. Right? **Right?**
+then that tells you that check is still being processed (e.g., the tests are still
+being run) so you'll need to wait a bit for the check to finish. The checks can
+take a few minutes, depending on their complexity and how busy the various services
+are. That's one of several reasons you should usually run the tests locally before
+you push.
+
+If any of the checks fail (give you a red x), then you probably want to click on
+"Details" by that check to learn more about what might have failed.
+
+* If the tests failed, it's likely that you either didn't structure your
+  greeting correctly, or you didn't put it in the correct place so that all
+  greetings are in alphabetical order. You should probably run the tests again
+  on your computer, where you'll get more information on which test failed and
+  why. If you're unsure how to proceed _definitely ask for some help!_
+* If either LGTM or Better Code Hub is grumpy, definitely click the "Details"
+  link, and you'll see more information about what they're not happy about.
+  If you understand the problem and how to possibly fix it, that's great.
+  If you don't, however, that's totally fine and you should definitely ask
+  for a hand!
+
+##### You might need to merge in changes from `main`
+
+Another thing that will block merging a pull request is if there have been
+changes made to `main` that you don't have in your pull request branch. Consider,
+for example, the case where Pat and Chris both create branches from `main`
+at about the same time. They both add their greetings, push their work, and
+create their pull requests. Let's imagine that (for whatever reason) Pat gets
+their changes merged into `main` a little before Chris. Now `main` will have
+Pat's changes, but those won't be in Chris's code; this will block Chris from
+merging into `main`.
+
+What does that look like on GitHub?
+
+One option is that When Chris looks at their pull request on GitHub they'll
+see a message that's something like:
+
+> This branch is out-of-date with the base branch
+
+In that case GitHub can merge the changes from `main` into Chris's branch
+automatically. If Chris is OK with that option, they can just press the
+"Update branch" button, and GitHub will do that merge.
+
+The more complicated option is when GitHub can't automatically merge the
+changes from `main` into Chris's branch, because there's a _merge conflict_.
+Merge conflicts happen when two branches both change the same bit of the code,
+and `git` isn't able to determine how to combine those changes. If that
+happens the message will be something like:
+
+> This branch has conflicts that must be resolved
+
+Here, if Chris clicks the "Resolve conflicts" button, GitHub will put Chris in
+a (web) editor where they can resolve the conflicts. That editor will likely
+have a section that looks something like:
+
+```java
+<<<<<<< chris-greeting
+    builder.append(chrisSaysHello());
+=======
+    builder.append(patSaysHello());
+>>>>>>> main
```

Comment:
Maybe point out that at the end, all the special `<<<<<<< chris-greeting`, `=======`, and `>>>>>>> main` lines should be removed.

Reply:
Totally! I also mean to just explain that weird notation more and obvious missed that in my rush.